docs/docsite/rst/dev_guide/migrating_roles.rst

```diff
@@ -0,0 +1,345 @@
+
+.. _migrating_roles:
+
+***************************************
+Migrating Roles to Roles in Collections
+***************************************
+
+You can migrate an existing standalone role into a collection-based role. With Ansible collections, you can create and distribute Ansible content that can contain one or many roles, modules or plugins (see :ref:`developing_collections` for details on collections).
+
+.. note::
+
+	Roles in collections cannot have hyphens in the role name. Rename any such roles to use underscores instead.
+
+
+.. contents::
+   :local:
+   :depth: 1
+
+.. _roles_in_collections:
+
+Comparing standalone roles to collection roles
+===============================================
+
+:ref:`Standalone roles <playbooks_reuse_roles>` have the following directory structure:
+
+.. code-block:: bash
+  :emphasize-lines: 5,7,8
+
+    role/
+    ├── defaults
+    ├── files
+    ├── handlers
+    ├── library
+    ├── meta
+    ├── module_utils
+    ├── [*_plugins]
+    ├── tasks
+    ├── templates
+    ├── tests
+    └── vars
+
+
+The highlighted directories above will change when you migrate to a collection-based role. The collection directory structure includes a :file:`roles/` directory:
+
+.. code-block:: bash
+
+    namespace/
+    └── name/
+      ├── docs/
+      ├── galaxy.yml
+      ├── plugins/
+      │   ├── modules/
+      │   │   └── module1.py
+      │   ├── inventory/
+      │   └── .../
+      ├── README.md
+      ├── roles/
+      │   ├── role1/
+      │   ├── role2/
+      │   └── .../
+      ├── playbooks/
+      │   ├── files/
+      │   ├── vars/
+      │   ├── templates/
+      │   └── tasks/
+      └── tests/
+
+A collection can contain one or more roles in the :file:`roles/` directory and these are almost identical to standalone roles with the following exceptions:
+
++----------------------------------------+------------------------------------------------------------------------------+
+| standalone role                        | collection role                                                              |
++========================================+==============================================================================+
+| :file:`library/` with custom modules   | - Move to :file:`plugins/modules/`                                           |
+|                                        | - Use :abbr:`FQCN (Fully Qualified Collection Name)` external to collection  |
```

Comment:
Additionally "external to collection" doesn't make sense to me—how is the FQCN different whether I'm in the collection or external to it? Isn't it always "namespace.name"?

One other question I ponder is what about collections that are not meant for Galaxy/Automation Hub—do they still need to have some sort of made-up namespace and name? Or can I have a collection called "stuff" that is not namespaced? (Haven't tested this, but wondering—since standalone roles can exist without a namespace.)

Reply:
I took out the table as it wasn't proving a good summary of what needs to happen. I also described  FQCN upfront. Not sure how to handle what you pondered in the end there about non-galaxy collections....